### ISSUE TITLE
Http/Client/Provider/Curl.php ignores full response default option

### ISSUE THREAD

**Issue by web-ted** [phalcon/incubator#687]
When making post request using http client in 3.0.x production releases a full body is returned by default. In master branch the issue seems fixed. The issue also doesn't appear in 2.0.x releases.
This breaks all http requests that use the body to retreive the response as it is.

**Comment by sergeyklay** [comment]
@web-ted Could you please explain a bit more. Provide simple script to reproduce, expected and actual behavior. 
Thanks

**Comment by sergeyklay** [comment]
Cc: @cottton

**Comment by web-ted** [comment]
$provider = Request::getProvider();
        $provider = $this->provider;
        $provider->setBaseUri('http://example.hostname.com:3325');
        $provider->header->set('Content-Type', 'application/x-www-form-urlencoded');
        $provider->setTimeout(10);
        $provider->setConnectTimeout(6);

        $responseSt = $provider->post('/test/add', 'option1=testOptionVal');

        echo $responseSt->body;

This returns the full body by default although it seems that false is the default behavior.

**Comment by web-ted** [comment]
To be more specific check this prototype from Curl class:

      private function send($customHeader = array(), $fullResponse = false);

This assumes that $fullResponse is false (by default) and the actual resposne includes full body response. It seems that something is happening while calling curl and setting its options. This doesn't happen to master branch  but it can be reproduced in 3.0.1.

**Comment by Ark4ne** [comment]
The problem is currently present on 3.0.1.

But, on 3.0.x-dev, it's worse.
I think @sergeyklay that you applied two different patch of the problem.

On 3.0.x-dev, the start of the response is truncated by the length of the header.

```php
$provider = Request::getProvider();

$response = $provider->get('http://www.google.com');

// hypothetically
// {header} = 'header'
// {body} = "sponse body"

$response->body === 'sponse body';
```

The fix I had proposed was to set `CURLOPT_HEADER = false`. Because using CURLOPT_HEADERFUNCTION, we do not need to retrieve the header in the curl response. It is recovered in all cases via CURLOPT_HEADERFUNCTION.

However, there is another "fixed", which removed the header from the reply, initially recover with CURLOPT_HEADER.
Only this increases the number of operation, and the memory used.

And currently there is a mix of both solutions on 3.0.x.

I put a link, below, with the functional code of Curl.php.
@web-ted if you can check.

[Curl.zip](https://github.com/phalcon/incubator/files/589796/Curl.zip)

This works fine for me with the patch
```php 
$provider = Request::getProvider();

$response = $provider->get('http://www.google.com', [], [], true);
```

**Comment by benjamineukles** [comment]
I think there is a problem with the solution given by @Ark4ne 
I was in version 3.0.1 and then ran a composer update, which has put the incubator to its lastest stable version 3.0.2.
After that, all my curl calls failed. When I've analyzed the response of the curl, i was that it was cut... And this is because in the version 3.0.2 the CURLOPT_HEADER option is set to false by default.

In the send method, the second parameter "$fullResponse" is set to "false" by default, but the handling of the response content is wrong.

```php
if ($fullResponse) {
            $response->body = $content;`
        } else {
            `$response->body = substr($content, strlen($this->responseHeader));
        }
```

It acts like there is always the header content in the curl response content, but it's wrong since CURLOPT_HEADER is set to false by default...

The solution, I think, is to init options like this:
```php
private function initOptions()
    {
        $this->setOptions([
            CURLOPT_RETURNTRANSFER  => true,
            CURLOPT_AUTOREFERER     => true,
            CURLOPT_FOLLOWLOCATION  => true,
            CURLOPT_MAXREDIRS       => 20,
            CURLOPT_HEADER          => true,
            CURLOPT_PROTOCOLS       => CURLPROTO_HTTP | CURLPROTO_HTTPS,
            CURLOPT_REDIR_PROTOCOLS => CURLPROTO_HTTP | CURLPROTO_HTTPS,
            CURLOPT_USERAGENT       => 'Phalcon HTTP/' . self::VERSION . ' (Curl)',
            CURLOPT_CONNECTTIMEOUT  => 30,
            CURLOPT_TIMEOUT         => 30,
        ]);
    }
```
with  **CURLOPT_HEADER => true,**

It worked for me.

**Comment by kowach** [comment]
Problem is that this [line](https://github.com/phalcon/incubator/blob/master/Library/Phalcon/Http/Client/Provider/Curl.php#L157)  where `$fullResponse` is false by default and this [line](https://github.com/phalcon/incubator/blob/master/Library/Phalcon/Http/Client/Provider/Curl.php#L80) where `CURLOPT_HEADER => false`.   So `$content` has only body, no header.
While `$this->responseHeader` is automagically filled by CURLOPT_HEADERFUNCTION. 

By default else is executed:

        if ($fullResponse) {
            $response->body = $content;
        } else {
            $response->body = substr($content, strlen($this->responseHeader));
        }

So body will be cut by first xx characters.

Workaround: `$provider->setOption(CURLOPT_HEADER, true);`

Solution: remove if else condition and just leave `$response->body = $content;`. Add `if ($fullResponse) { $this->setOption(CURLOPT_HEADER, true); }` before curl_exec.

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch. Thanks
